app/azure/_module.js

```diff
@@ -22,13 +22,26 @@ angular
               $state.go('portainer.home');
               return;
             }
+            var preEndpoint = EndpointProvider.endpoints();
             try {
               EndpointProvider.setEndpointID(endpoint.Id);
               EndpointProvider.setEndpointPublicURL(endpoint.PublicURL);
               EndpointProvider.setOfflineModeFromStatus(endpoint.Status);
               await StateManager.updateEndpointState(endpoint, []);
             } catch (e) {
               Notifications.error('Failed loading environment', e);
+              if (preEndpoint) {
+                EndpointProvider.setEndpoints(preEndpoint);
+              }
+              if (preEndpoint.Id) {
+                EndpointProvider.setEndpointID(preEndpoint.Id);
+              }
+              if (preEndpoint.PublicURL) {
+                EndpointProvider.setEndpointPublicURL(preEndpoint.PublicURL);
+              }
+              if (preEndpoint.Status) {
+                EndpointProvider.setOfflineModeFromStatus(preEndpoint.Status);
+              }
               $state.go('portainer.home', {}, { reload: true });
```

Comment:
won't this work? and cleaner?
```suggestion

            try {
              await StateManager.updateEndpointState(endpoint, []);
               EndpointProvider.setEndpointID(endpoint.Id);
              EndpointProvider.setEndpointPublicURL(endpoint.PublicURL);
              EndpointProvider.setOfflineModeFromStatus(endpoint.Status);
            } catch (e) {
              Notifications.error('Failed loading environment', e);
              $state.go('portainer.home', {}, { reload: true });
```

Reply:
No, this won't work... I tried your code in my local environment...It did not...

Reply:
I have tried another method and verified on my env